### DISCUSSION TITLE
Software architecture question

### DISCUSSION THREAD

**Discussion by thosil** [Ulauncher/Ulauncher#1397] [General]
Hello, I'm looking at the code and was surprised that a part of the app was written in Vue. I dig a little bit to understand that it's for preference window and you start an api server.
Why did you choose that way? I mean, the rest of the app is already a python Gtk app, so what's the motivation?
Thanks

**Comment by friday** [comment]
https://github.com/Ulauncher/Ulauncher/issues/1257
